packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CopyView/useCopyView.ts

```diff
@@ -2,20 +2,29 @@ import { useState } from 'react';
 
 import { isFunction } from '@aws-amplify/ui';
 
-import { copyHandler } from '../../../actions/handlers';
+import { LocationData, copyHandler } from '../../../actions/handlers';
 import { useProcessTasks } from '../../../tasks';
 import { useGetActionInput } from '../../../providers/configuration';
 import { useStore } from '../../../providers/store';
-import { getActionViewDisabledButtons } from '../utils';
-import { getTaskCounts } from '../../../controls/getTaskCounts';
 import { getDestinationListFullPrefix } from '../utils/getDestinationPickerDataTable';
 import { CopyViewState } from './types';
 
-export const useCopyView = ({
-  onExit: _onExit,
-}: {
-  onExit?: () => void;
+const getInitialDestinationList = (key: string, prefix?: string) =>
+  // handle root bucket access grant
+  key === ''
+    ? ['']
+    : // handle subfolder inside root access grant
+    key && prefix == ''
```

Comment:
Purposefully using `==`?

Reply:
Just moved this from inside the useCopyView code, will most likely be removed in a follow up

Reply:
Ok we should make sure it's removed because this is technically wrong albeit.. seems unlikely prefix will ever be `false`